hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -477,7 +482,16 @@ private void initializeFromFilesystem(String initializationTime, List<MetadataPa
       HoodieData<HoodieRecord> records = fileGroupCountAndRecordsPair.getValue();
       bulkCommit(instantTimeForPartition, partitionName, records, fileGroupCount);
       metadataMetaClient.reloadActiveTimeline();
-
+      if (partitionType == COLUMN_STATS) {
+        // initialize Col Stats
+        // if col stats, lets also update list of columns indexed if changed.
+        updateColumnsToIndexWithColStats(columnsToIndex);
```

Comment:
If we're anyway going to check and update post commit, why doing it here?

Reply:
essentially, for regular commits, both mdt write and data table write has to complete and we update the index defn. 
but for initialization, its a special commit, once mdt commit is completed, readers are eligible to read the data. and hence its better to keep the index defn updated.